src/components/dashboard/JoinGroup.jsx

```diff
@@ -0,0 +1,67 @@
+import React from 'react';
+import { Modal, Input } from 'antd';
+import Button from '../Button.jsx';
+import '../../scss/join-group.scss';
+// import { func } from 'prop-types';
+
+function JoinGroup() {
+  const [visible, setVisible] = React.useState(false);
+  const [confirmLoading, setConfirmLoading] = React.useState(false);
+  const [errorText, setErrorText] = React.useState('');
+  const [code, setCode] = React.useState('');
+
+  function showModal() {
+    setCode('');
+    setVisible(true);
+  }
+
+  function handleCancel() {
+    setCode('');
+    setVisible(false);
+  }
+
+  function submitCode(event) {
+    event.preventDefault();
+    setConfirmLoading(true);
+
+    try {
+      // TODO: Hook up join group api
+      // Dummy time out to simulate async
+      setTimeout(() => {
+        setCode('');
+        setVisible(false);
+        setConfirmLoading(false);
+      }, 1000);
+    } catch (e) {
+      setErrorText(e);
+    }
+  }
+
+  return (
+    <div className='center'>
+      <Button onClick={showModal}>Join Group</Button>
+      <Modal
+        title='Join group'
+        visible={visible}
+        onOk={submitCode}
+        confirmLoading={confirmLoading}
+        onCancel={handleCancel}
+        okText='Join'
+      >
+        <form onSubmit={submitCode}>
+            <p>Enter an eight character group code.</p>
+            <Input
+              placeHolder='Example: xJwY394p'
+              onChange={(c) => { setCode(code + c.nativeEvent.data); }}
```

Comment:
I think using `event.target.value` would be a bit cleaner in this case. For example
```js
onChange={event => setCode(event.target.value)}
```
That would give you the current text of the input so far instead of each individual character.